tests/integration/compile/compile-es6-spec.js

```diff
@@ -0,0 +1,355 @@
+                'var a = _ref[0];\n' +
+                'var b = _ref[2];',
+                done
+            ))
+            .catch(throwError);
+
+    });
+
+    it('Given ES6 code with destructured variable assignment with an object, compiles to ES5.', function(done) {
+
+        compile('var {a, b} = {a:1, b:2};')
+            .then(assertResultIs(
+                '"use strict";\n' +
+                '\n' +
+                'var _a$b = { a: 1, b: 2 };\n' +
+                'var a = _a$b.a;\n' +
+                'var b = _a$b.b;',
+                done
+            ))
+            .catch(throwError);
+
+    });
+
+    it('Given ES6 code with default function parameters, compiles to ES5.', function(done) {
+
+        compile('function f(x = 1) { return x; }')
+            .then(assertResultIs(
+                '"use strict";\n' +
+                '\n' +
+                'function f() {\n' +
+                '  var x = arguments.length <= 0 || arguments[0] === undefined ? 1 : arguments[0];\n' +
+                '  return x;\n' +
+                '}',
+                done
+            ))
+            .catch(throwError);
+
+    });
+
+    it('Given ES6 code with rest parameter, compiles to ES5.', function(done) {
+
+        compile('function f(x, ...y) { return x + y.length; }')
+            .then(assertResultIs(
+                '"use strict";\n' +
+                '\n' +
+                'function f(x) {\n' +
+                '  return x + (arguments.length - 1);\n' +
+                '}',
+                done
+            ))
+            .catch(throwError);
+
+    });
+
+    it('Given ES6 code with let variables, compiles to ES5.', function(done) {
+
+        compile('let x = 5;')
+            .then(assertResultIs(
+                '"use strict";\n' +
+                '\n' +
+                'var x = 5;',
+                done
+            ))
+            .catch(throwError);
+
+    });
+
+    it('Given ES6 code with const variables, compiles to ES5.', function(done) {
+
+        compile('const x = 5;')
+            .then(assertResultIs(
+                '"use strict";\n' +
+                '\n' +
+                'var x = 5;',
+                done
+            ))
+            .catch(throwError);
+
+    });
+
+    it('Given ES6 code with function shorthand, compiles to ES5.', function(done) {
+
+        compile('var x = { foo() { return 1; } };')
+            .then(assertResultIs(
+                '"use strict";\n' +
+                '\n' +
+                'var x = {\n' +
+                '  foo: function foo() {\n' +
+                '    return 1;\n' +
+                '  }\n' +
+                '};',
+                done
+            ))
+            .catch(throwError);
+
+    });
+
+    it('Given ES6 code with async/await, compiles to ES5.', function(done) {
+
+        compile(
+                'async function foo() {\n' +
+                '    return await new Promise(function(resolve, reject) {\n' +

```

Comment:
We didn't previously support this, but I added it in since it works in IE8 and IE9. `async`/`await`, just like C#!